src/java/com/sarality/app/view/list/ListViewItemActionPerformer.java

```diff
@@ -0,0 +1,93 @@
+package com.sarality.app.view.list;
+
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.LinearLayout;
+import android.widget.ListView;
+
+import com.sarality.app.view.action.Action;
+import com.sarality.app.view.action.TriggerType;
+
+/**
+ * Performs the Action when an event on an Item in a ListView is triggered.
+ * 
+ * @author abhideep@ (Abhideep Singh)
+ */
+class ListViewItemActionPerformer implements AdapterView.OnItemClickListener, AdapterView.OnItemLongClickListener,
+    View.OnClickListener, View.OnLongClickListener {
+
+  private final static String LIST_POSITION_TAG = "ListPosition";
```

Comment:
There are two way to accomplish this

1) If the Action is created in the doRender then the data can be passed to the Action in the constructor itself and then a call made to registerAction.

2) The View is passed to the Action via the ViewActionContext. The data associated with the View should be extracted from the view itself where it can be saved in the form of Tag values. 

How have we done this in the past?